controllers/osbuild_controller.go

```diff
@@ -47,16 +82,336 @@ type OSBuildReconciler struct {
+	checkGpg := true
+	rhsm := true
+	return []composer.Repository{
+		{
+			Baseurl:  &baseosUrl,
+			CheckGpg: &checkGpg,
+			Gpgkey:   &baseosgpgkey,
+			Rhsm:     &rhsm,
+		},
+		{
+			Baseurl:  &appstreamUrl,
+			CheckGpg: &checkGpg,
+			Gpgkey:   &appstreamgpgkey,
+			Rhsm:     &rhsm,
+		},
+	}
+}
+func (r *OSBuildReconciler) createImageRequest(targetImage *osbuildv1alpha1.TargetImage, targetImageType osbuildv1alpha1.TargetImageType) *composer.ImageRequest {
+	uploadOptions := composer.UploadOptions(composer.AWSS3UploadOptions{Region: ""})
+
+	// TODO - add repositories to osbuildConfig and osbuildConfigTemplate types
+	imageRequest := composer.ImageRequest{
+		Architecture:  string(targetImage.Architecture),
+		ImageType:     composer.ImageTypes(targetImageType),
+		UploadOptions: &uploadOptions,
+		Repositories:  r.buildRepositories(),
+	}
+	if targetImage.OSTree != nil {
+		imageRequest.Ostree = (*composer.OSTree)(targetImage.OSTree.DeepCopy())
+	}
+	return &imageRequest
+}
+
+func (r *OSBuildReconciler) createCustomizations(osbuildCustomizations *osbuildv1alpha1.Customizations) *composer.Customizations {
+	if osbuildCustomizations == nil {
+		return &composer.Customizations{}
+	}
+
+	composerCustomizations := composer.Customizations{}
+	if osbuildCustomizations.Users != nil && len(osbuildCustomizations.Users) > 0 {
+		var users []composer.User
+		for _, cstmzUser := range osbuildCustomizations.Users {
+			user := cstmzUser.DeepCopy()
+			users = append(users, (composer.User)(*user))
+		}
+
+		composerCustomizations.Users = &users
+	}
+
+	if osbuildCustomizations.Services != nil {
+		var services struct {
+			Disabled *[]string `json:"disabled,omitempty"`
+			Enabled  *[]string `json:"enabled,omitempty"`
+		}
+
+		if osbuildCustomizations.Services.Enabled != nil {
+			services.Enabled = &osbuildCustomizations.Services.Enabled
+		}
+		if osbuildCustomizations.Services.Disabled != nil {
+			services.Disabled = &osbuildCustomizations.Services.Disabled
+		}
+		composerCustomizations.Services = &services
+	}
+	if osbuildCustomizations.Packages != nil {
+		composerCustomizations.Packages = &osbuildCustomizations.Packages
+	}
+
+	return &composerCustomizations
```

Comment:
You have the same issue with the Customization structure as a whole since it is not mandatory either. So, if Customizations is empty (meaning all the `if` statements were `false`), you should return `nil`